### DISCUSSION TITLE
How to force png output be 32 bpp?

### DISCUSSION THREAD

**Discussion by tim-tim** [ImageMagick/ImageMagick#2377] [Help] [answered]
While compositing two PNG 32 BPP images with command like
`magick left.png right.png -composite result.png`

I get png file with 8 BPP, but I need png with 32 BPP depth.

p.s.: At the same time saving to tiff output gives 32BPP depth.

ImageMagick 7.0.10-25 Q8 x64 (Windows)

**Comment by fmw42** [comment, marked as answer]
Try

`magick left.png right.png -composite PNG32:result.png`

Is the left image grayscale or binary?

**Reply by tim-tim** [reply]
> Try
> 
> `magick left.png right.png -composite PNG32:result.png`
> 
> Is the left image grayscale or binary?

Both left.png and right.png - "colored" sRGB 32bpp. But they have only one color drawings on transparent backgound. It seems like IM trying to optimize output by default.

Wow, it works! Thank you very much! 
I'm still in confusion with all that parameters voodoo magick (that PNG32:result.png syntax)

**Comment by fmw42** [comment]
Most formats respect -type truecoloralpha for that, except PNG. Not sure why PNG is different.

**Comment by snibgo** [comment]
> I get png file with 8 BPP, but I need png with 32 BPP depth.

That might be the same thing. IM generally counts bits per channel per pixel. Other programs often count total bits per pixel.

8 bits per channel per pixel for 4 channels equals 32 total bits per pixel.

But sometimes "8 BPP" means the image has a palette with up to 256 colours, where each colour might have 32 bits.

Yes, it is confusing.
